rust/src/dataset.rs

```diff
@@ -440,6 +440,85 @@ impl Dataset {
         Self::write_impl(batches, uri, params).await
     }
 
+    /// Append to existing [Dataset] with a stream of [RecordBatch]s
+    ///
+    /// Returns void result or Returns [Error]
+    pub async fn append(
+        &mut self,
+        batches: impl RecordBatchReader + Send + 'static,
+        mut params: Option<WriteParams>,
+    ) -> Result<()> {
+        let batches = Box::new(batches);
+
+        // Check params are correct
+        if params.is_none() {
+            let mut _params = WriteParams::default();
+            _params.mode = WriteMode::Append;
+            params = Some(_params);
+        } else {
+            if !matches!(params.as_ref().unwrap().mode, WriteMode::Append) {
+                return Err(Error::InvalidOperation {
+                    message: format!("Invalid create mode on append operation."),
+                });
+            }
```

Comment:
I think it's reasonable to just ignore write mode altogether, or force it to append. Seems unnecessarily complex to have a method called `append()` and then validate again that the user provided the correct write mode.

Reply:
Sure, that makes sense. I thought so on similar lines but wanted to be sure. I'll make the appropriate changes.